src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,322 @@
+package kitchenpos.application;
+
+import kitchenpos.dao.MenuDao;
+import kitchenpos.dao.MenuGroupDao;
+import kitchenpos.dao.MenuProductDao;
+import kitchenpos.dao.ProductDao;
+import kitchenpos.domain.Menu;
+import kitchenpos.domain.MenuProduct;
+import kitchenpos.domain.Product;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("MenuService 클래스 테스트")
+public class MenuServiceTest {
+    @Mock
+    private MenuDao menuDao;
+    @Mock
+    private MenuGroupDao menuGroupDao;
+    @Mock
+    private MenuProductDao menuProductDao;
+    @Mock
+    private ProductDao productDao;
+    @InjectMocks
+    private MenuService menuService;
+
+    @Nested
+    @DisplayName("create 메서드 테스트")
+    public class CreateMethod {
+        @Nested
+        @DisplayName("메뉴 등록 성공")
+        public class Success {
+            @Test
+            public void testCase() {
+                // given
+                final Menu mockMenu = setup();
+
+                // when
+                final Menu createdMenu = menuService.create(mockMenu);
+
+                // then
+                assertAll(
+                        () -> assertThat(createdMenu.getName()).isEqualTo(mockMenu.getName()),
+                        () -> assertThat(createdMenu.getPrice()).isEqualTo(mockMenu.getPrice()),
+                        () -> assertThat(createdMenu.getMenuGroupId()).isEqualTo(mockMenu.getMenuGroupId()),
+                        () -> assertAllMenuProducts(createdMenu, mockMenu)
+                );
+            }
+
+            private Menu setup() {
+                final long menuGroupId = 1L;
+                Mockito.when(menuGroupDao.existsById(menuGroupId)).thenReturn(true);
+                final List<Long> productsId =
+                        Stream.of(11, 12, 13).map(Long::new).collect(Collectors.toList());
+                final List<Product> products = productsId
+                        .stream()
+                        .map(productId -> {
+                            final Product product = new Product();
+                            product.setId(productId);
+                            product.setPrice(BigDecimal.valueOf(100 * productId));
```

Comment:
```suggestion
                            final Product product = new Product(productId, BigDecimal.valueOf(100 * productId));
```

도메인 클래스의 생성자를 수정하면 더 간략하게 작성할수 있겠네요 😀